org.eclipse.jdt.debug.ui/ui/org/eclipse/jdt/internal/debug/ui/console/JavaStackTraceHyperlink.java

```diff
@@ -183,18 +199,180 @@ public IStatus runInUIThread(IProgressMonitor monitor) {
+														int indexOfClosing = innerMethod.toString().indexOf(')');
+														int indexOfStart = innerMethod.toString().indexOf(innerMethod.getElementName());
+														String methodName = innerMethod.toString().substring(indexOfStart, indexOfClosing
+																+ 1).replaceAll(" ", ""); //$NON-NLS-1$//$NON-NLS-2$
+														int paramCount = firstMethod.substring(firstMethod.indexOf('(')
+																+ 1, firstMethod.lastIndexOf(')')).split(",").length; //$NON-NLS-1$
+														if (methodName.equals(firstMethod)) {
+															exactMatchesFiltered.add(obj);
+														} else if (firstMethodName.equals(innerMethod.getElementName())
+																&& paramCount == innerMethod.getNumberOfParameters()) {
+															StringBuilder s = new StringBuilder(innerMethod.getElementName());// Further mining from
+																																// fully
+																																// qualified
+																																// parameter names in
+																																// method signature
+															s.append('(');
+															String[] params = methodName.split(","); //$NON-NLS-1$
+															for (String block : params) {
+																if (block.contains("...")) { //$NON-NLS-1$ Parameter is var args
+																	if (params.length > 1) {
+																		String sub1 = block.substring(0, block.indexOf("...")); //$NON-NLS-1$
+																		sub1 = sub1.substring(sub1.lastIndexOf('.') + 1);
+																		s.append(sub1);
+																		s.append('.');
+																		s.append('.');
+																		s.append('.');
+																		s.append(',');
+																	}
+																} else {
+																	if (block.indexOf('.') == -1) {
+																		s.append(block.substring(block.lastIndexOf('(') + 1));
+																		s.append(',');
+																	} else {
+																		s.append(block.substring(block.lastIndexOf('.') + 1));
+																		s.append(',');
+																	}
+																}
+															}
+
+															s.deleteCharAt(s.length() - 1);
+
+															if (s.charAt(s.length() - 1) != ')') {
+																s.append(')');
+															}
+															if (s.toString().equals(firstMethod)) {
+																exactMatchesFiltered.add(obj);
+															}
+														}
+													}
+												}
+
+											}
+
+										}
+									}
+
+								} catch (JavaModelException e) {
+									JDIDebugUIPlugin.log(e);
+								}
+							}
+						}
+					}
+					int line = lineNumber + 1;
+					if (exactMatchesFiltered.size() == 1) {
+						processSearchResult(exactMatchesFiltered.get(0), typeName, lineNumber);
+						return Status.OK_STATUS;
+					}
+					if (exactMatchesFiltered.size() == 2) {
+						try { // Occurred only in child eclipse
+							if (exactMatchesFiltered.get(0) instanceof BinaryType && exactMatchesFiltered.get(1) instanceof BinaryType) {
```

Comment:
can you get rid of the compiler warnings here? `type 'BinaryType' is not API`
Also its handsome to use pattern matching instanceof where possible:
`(x instanceof IType y) y.method()`

Reply:
Sure Jukzi, will do like you suggested.